mllib/src/test/scala/org/apache/spark/ml/clustering/BisectingKMeansSuite.scala

```diff
@@ -51,6 +54,23 @@ class BisectingKMeansSuite
     assert(copiedModel.hasSummary)
   }
 
+  test("SPARK-16473: Verify Bisecting K-Means does not fail in edge case where" +
+    "one cluster is empty after split") {
+    val bkm = new BisectingKMeans().setK(k).setMinDivisibleClusterSize(4).setMaxIter(4)
+
+    assert(bkm.getK === k)
+    assert(bkm.getFeaturesCol === "features")
+    assert(bkm.getPredictionCol === "prediction")
+    assert(bkm.getMaxIter === 4)
+    assert(bkm.getMinDivisibleClusterSize === 4)
+    // Verify fit does not fail on very sparse data
```

Comment:
It's not clear to me that this unit test actually tests the issue fixed in this PR.  Is there a good way to see why it would?  If not, then it would be great to write a tiny dataset by hand which would trigger the failure.